rt_waterfall_D3S.py

```diff
@@ -0,0 +1,112 @@
+from auxiliaries import set_verbosity
+import time
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+class Rt_Waterfall_D3S(object):
+    """
+    Class for running the D3S in real-time waterfall mode
+    """
+    
+    def __init__(self, 
+                 manager=None, 
+                 verbosity=1,
+                 logfile=None,
+                ):
+        
+        self.v = verbosity
+        if manager and logfile is None:
+            set_verbosity(self, logfile=manager.logfile)
+        else:
+            set_verbosity(self, logfile=logfile)
+            
+        self.manager = manager
+        
+        self.interval = manager.interval
+        
+        self.queuelength = None
+        self.image = None
+    
+    def get_data(self, spectra):
+        '''
+        Transfers spectra data to 2 queues that were designed for 
+        real time waterfall mode.
+        Call rebin spectra in this method.
+        '''
+        new_spectra = self.rebin(spectra)
+        self.manager.wqueue2.append(new_spectra)
+        self.queuelength = len(self.manager.wqueue2)
+        for i in self.manager.wqueue2:
+            self.manager.wqueue1.append(i)
```

Comment:
Why do you need two queues? My initial reading is that you're just passing spectra from one queue into the other...